shared/src/types.rs

```diff
@@ -437,7 +450,11 @@ pub struct ListenPortOpts {
 
 #[derive(Debug, Clone, PartialEq, Eq, Args)]
 pub struct OverrideEndpointOpts {
-    /// The listen port you'd like to set for the interface
+    /// The external endpoint that you'd like the innernet server to broadcast
+    /// to other peers. The IP address may be unspecified, in which case the
+    /// server will try to resolve it based on its most recent connection.
+    /// The port will still be used even if you decide to use the unspecified
+    /// IP address.
     #[clap(short, long)]
     pub endpoint: Option<Endpoint>,
```

Comment:
Before this I didn't even know what "unspecified IP address" exactly means. Shall we give an example like `0.0.0.0:1234`?